customers/schema/queries.py

```diff
@@ -179,6 +216,17 @@ def resolve_berth_profiles(self, info, **kwargs):
 
         qs = CustomerProfile.objects
 
+        # Check if Helsinki Profiles filters are used in the query, if yes, query the
+        # profile ids first from there
+        if not set(kwargs.keys()).isdisjoint(set(HELSINKI_PROFILES_FILTERS)):
+            profile_token = kwargs.pop("api_token", None)
+            if not profile_token:
+                raise VenepaikkaGraphQLError(
+                    "Cannot filter by Helsinki Profile fields without API Token"
+                )
+            ids = _get_ids_from_profile_service(kwargs, profile_token)
+            qs = qs.in_bulk(id_list=[ids])
```

Comment:
in_bulk does not return a query set, but a dict! https://docs.djangoproject.com/en/4.0/ref/models/querysets/#in-bulk. I implemented this with Case-When in #648 as @charn proposed